qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -396,99 +575,178 @@ def solve(
             t_span, y0, signals
         )
 
-        all_results = [
-            self._solve(
-                t_span=current_t_span,
-                y0=current_y0,
-                signals=current_signals,
+        all_results = None
+        if (
+            Array.default_backend() == "jax"
+            and is_jax_method(kwargs.get("method", ""))
+            and all(isinstance(x, (Schedule, ScheduleBlock)) for x in signals_list)
+        ):
+            all_results = self._solve_schedule_list_jax(
+                t_span_list=t_span_list,
+                y0_list=y0_list,
+                schedule_list=signals_list,
                 convert_results=convert_results,
                 **kwargs,
             )
-            for current_t_span, current_y0, current_signals in zip(
-                t_span_list, y0_list, signals_list
+        else:
+            all_results = self._solve_list(
+                t_span_list=t_span_list,
+                y0_list=y0_list,
+                signals_list=signals_list,
+                convert_results=convert_results,
+                **kwargs,
             )
-        ]
-
-        # replace copy of original signals for deprecated behavior
-        self.model.signals = original_signals
 
         if multiple_sims is False:
             return all_results[0]
 
         return all_results
 
-    def _solve(
+    def _solve_list(
         self,
-        t_span: Array,
-        y0: Union[Array, QuantumState, BaseOperator],
-        signals: Optional[Union[List[Signal], Tuple[List[Signal], List[Signal]]]] = None,
-        convert_results: Optional[bool] = True,
+        t_span_list: List[Array],
+        y0_list: List[Union[Array, QuantumState, BaseOperator]],
+        signals_list: Optional[
+            Union[List[Schedule], List[List[Signal]], List[Tuple[List[Signal], List[Signal]]]]
+        ] = None,
+        convert_results: bool = True,
         **kwargs,
-    ) -> OdeResult:
-        """Helper function solve for running a single simulation."""
-        # convert types
-        if isinstance(y0, QuantumState) and isinstance(self.model, LindbladModel):
-            y0 = DensityMatrix(y0)
+    ) -> List[OdeResult]:
+        """Run a list of simulations."""
 
-        y0, y0_cls, state_type_wrapper = initial_state_converter(y0)
+        all_results = []
+        for t_span, y0, signals in zip(t_span_list, y0_list, signals_list):
 
-        # validate types
-        if (y0_cls is SuperOp) and is_lindblad_model_not_vectorized(self.model):
-            raise QiskitError(
-                """Simulating SuperOp for a LindbladModel requires setting
-                vectorized evaluation. Set LindbladModel.evaluation_mode to a vectorized option.
-                """
+            if isinstance(signals, (Schedule, ScheduleBlock)):
```

Comment:
I don't think we should be allowing `ScheduleBlock` here as it may not have the concept of pulse instructions played at an absolute time. This is why the converter only allows `Schedule`.

Reply:
`ScheduleBlock` instances are ultimately converted to `Schedule`s before simulation is attempted. Do you think it makes sense to do this? I got the impression that users wouldn't manually be converting `ScheduleBlock` instances into `Schedule`s, so we should do it automatically here.

Reply:
Looking at this again I think it makes sense to do this conversion for the user. But I expect that this happens in only one place. Is that the case now?